### PR TITLE
feat(evals): add evaluation item metadata registry

### PR DESCRIPTION
**Pull request by Katsiarynakavaleuskaya**
## Summary

Adds a deterministic evaluation item metadata registry for psychometric readiness.

The Evaluation Science foundation now has item-level outcomes, sidecars, invariance/mutation fixtures, and canonical-fail negative controls (PR #1659). This PR adds the missing registry layer that maps every canonical eval item to stable metadata needed before future item weighting, IRT, or adaptive eval work.

**Registry:** 10 items (5 judgment + 5 RAG), covering all canonical_ids from both variant fixture sets, including canonical-fail negative control items.

## Scope

- `scripts/evals/eval_item_registry.py` — registry contract module (TypedDict schema, validation, loading, coverage checks)
- `data/evals/eval_item_metadata_registry.jsonl` — registry data (10 rows, one per fixture canonical_id)
- `tests/evals/test_eval_item_metadata_registry.py` — 15 deterministic tests
- `docs/evals/PULSEPLATE_EVAL_VALIDITY_CONTRACT.md` — new registry section
- `docs/evals/PULSEPLATE_RAG_RELEASE_GATES.md` — short note about RAG items in registry
- `docs/orchestration/JUDGMENT_ADJUDICATION_SUBLANE_PROTOCOL.md` — short note about judgment items in registry
- `docs/roadmap/BACKLOG_LEDGER.md` — P1 backlog item

## Non-goals

- No IRT implementation
- No psychometric scoring
- No adaptive evals
- No hybrid adjudication
- No tool-use reliability metrics
- No mechanistic evaluation
- No runtime/API/frontend/iOS/billing/OpenAPI/App Store changes
- No Claude / Opus runtime integration
- No `.claude/`
- No OpenRouter provider
- No MCP
- No LLM-generated metadata

## Validation

- [x] `python3 scripts/orchestration/check_preflight.py`
- [x] `python3 scripts/orchestration/check_agent_consistency.py`
- [x] `pytest -q tests/evals/test_eval_item_metadata_registry.py` (15/15 pass)
- [x] `pytest -q tests/evals/` (152/152 pass)
- [x] `pytest -q tests/test_judgment_eval.py tests/test_rag_release_gates_runner.py` (57/57 pass)
- [x] `git diff --check` (clean)
- [x] Pre-push hooks: black, ruff, mypy, pip-audit, pytest, bandit, docker build — all pass

## Pre-mortem Mitigations

- [x] Registry does not claim IRT scoring — docs state "psychometric-readiness only"
- [x] Registry covers every fixture canonical_id (10/10)
- [x] Registry has no orphan canonical_id rows
- [x] Expected decisions match canonical fixture rows (8 pass + 2 fail)
- [x] Difficulty bands are explicit labels, not calibrated psychometric estimates
- [x] No provider/model/network metadata (test 13 scans for forbidden strings)
- [x] No runtime or release-gate changes

## Security Notes

This PR is offline and repo-local. It introduces no credentials, no external model calls, no provider routing, no real user data, and no production runtime behavior.

## Decision Log

1. Item registry is a psychometric-readiness layer, not psychometric scoring.
2. IRT / item information modeling remains deferred.
3. Validity metrics remain unchanged.
4. RAG release-gate decisions remain canonical.
5. Judgment decisions remain canonical.
6. Opus is used only as an operator coding model and is not integrated into runtime.
7. PR is opened as normal PR, not draft.
8. Canonical-fail negative control items (PR #1659) are included with expected_decision="fail" and expected_score_band="fail".

## Deferred / Follow-ups

- Psychometrics / IRT item modeling — `docs/roadmap/BACKLOG_LEDGER.md#ledger-p1-eval-item-metadata-registry`
- Item weighting based on empirical runs
- Hybrid adjudication framework
- Tool-use reliability metrics
- Adaptive self-improving eval loop
- Mechanistic evaluation research lane

## Discussion Thread Pass

- [x] Discussion-thread pass completed
- [x] Fixed in commit mapping completed

### Fixed in Commit Mapping

- canonical artifact: docs/review/PR_1660_FIXED_MAPPING.md

## Split Justification

This PR exceeds 800 LoC because it contains a complete, self-contained work
package: registry module (234 lines) + JSONL data (10 lines) + test suite
(~490 lines) + docs (~100 lines). Splitting would break the bidirectional
coverage invariant that tests enforce between registry and data. All code
is offline eval infrastructure with no runtime dependencies.

## Merge Readiness

- [ ] Required CI green on current head
- [ ] `make verify` passed locally or raw failure output documented
- [ ] Review mapping artifact created
- [ ] Pre-mortem artifact created
- [ ] CodeRabbit / Sourcery / Cubic have no actionables
- [ ] Mandatory wait-window elapsed
- [ ] Final strict merge-readiness check passed

---
## Summary by cubic
Adds a deterministic eval item metadata registry that maps each canonical eval item to stable metadata for psychometric readiness. No scoring or release‑gate changes; review docs now include complete fixed‑mapping and merge‑readiness URL links.

- **New Features**
  - Registry module with schema, strict validation, loader, and coverage helpers; deterministic tests for schema, coverage, decision parity, and variant‑family coverage; docs updated in validity contract, RAG release‑gates, and judgment protocol.
  - Registry data in `data/evals/eval_item_metadata_registry.jsonl` for 10 `canonical_id`s (judgment + RAG), including anchors and canonical‑fail controls; records expected decisions, difficulty bands, and variant‑family coverage.
  - Review artifacts: premortem added and fixed‑mapping completed with review‑URL links.

- **Bug Fixes**
  - Hardened validation and tests: reject non‑dict/extra‑key input, strict type checks, empty‑list guard, cross‑lane `canonical_id` collision guard; guard submodule/qualified network imports; fail‑fast on duplicate canonical fixture rows.

<sup>Written for commit 95ce1a2397cbb5929026de51f46af6e7cca24539. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added a stable eval-item metadata registry containing 10 canonical items that record lane, domain, skill, difficulty, expected outcomes, and variant-family coverage.

* **Documentation**
  * Updated protocols, release-gate, adjudication, roadmap, and review docs to describe the registry, coverage rules, limitations, and next steps.

* **Tests**
  * Added deterministic tests enforcing schema, uniqueness, fixture-coverage parity, forbidden-content scans, and negative/edge-case validations.

### CODE REVIEW

File: tests/evals/test_eval_item_metadata_registry.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+                for alias in node.names:
+                    assert (
+                        alias.name not in _NETWORK_MODULES
+                    ), f"eval_item_registry.py imports network lib: {alias.name}"
+            elif isinstance(node, ast.ImportFrom):
+                if node.module is not None:
+                    top_module = node.module.split(".")[0]
+                    assert (
+                        top_module not in _NETWORK_MODULES
+                    ), f"eval_item_registry.py imports from network lib: {node.module}"
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/evals/test_eval_item_metadata_registry.py]: Disposition: FIXED
Commit: a8e5751ff
Evidence: `tests/evals/test_eval_item_metadata_registry.py:365-376` — ImportFrom guard now checks qualified names (`from urllib import request` → `urllib.request` matched against forbidden list).

File: tests/evals/test_eval_item_metadata_registry.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+    """Load canonical rows (variant_family == 'canonical') keyed by canonical_id."""
+    rows: dict[str, dict] = {}
+    with path.open("r", encoding="utf-8") as fh:
+        for line in fh:
+            stripped = line.strip()
+            if not stripped:
+                continue
+            raw = json.loads(stripped)
+            if raw.get("variant_family") == "canonical":
+                rows[raw["canonical_id"]] = raw
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/evals/test_eval_item_metadata_registry.py]: Disposition: FIXED in 35501f99d. _load_fixture_canonical_rows() now raises ValueError on duplicate canonical_id.
